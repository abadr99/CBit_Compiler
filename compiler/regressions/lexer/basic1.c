// RUN: cbit %s --keep_lexer=true 
// RUN: diff basic1.lex basic1.expected 

int main() {
    int n, i;
    unsigned long long fact = 1;
    printf("Enter an integer: ");
    scanf("%d", &n);

    // shows error if the user enters a negative integer
    if (n < 0)
        printf("Error! Factorial of a negative number doesn't exist. :(");
    else {
        for (i = 1; i <= n; ++i) {
            fact *= i;
        }
        printf("Factorial of %d = %llu", n, fact);
    }

    return 0;
}
