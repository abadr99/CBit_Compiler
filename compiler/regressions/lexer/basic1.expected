Token-Value                                                                     Token-Type
int                                                                             KeyWord
main                                                                            ID
(                                                                               Operator
)                                                                               Symbol
{                                                                               Operator
int                                                                             KeyWord
n                                                                               ID
,                                                                               Symbol
i                                                                               ID
;                                                                               Symbol
unsigned                                                                        KeyWord
long                                                                            KeyWord
long                                                                            KeyWord
fact                                                                            ID
=                                                                               Operator
1                                                                               Number
;                                                                               Symbol
printf                                                                          ID
(                                                                               Operator
"Enter an integer: "                                                            String
)                                                                               Symbol
;                                                                               Symbol
scanf                                                                           ID
(                                                                               Operator
"%d"                                                                            String
,                                                                               Symbol
&                                                                               Operator
n                                                                               ID
)                                                                               Symbol
;                                                                               Symbol
if                                                                              KeyWord
(                                                                               Operator
n                                                                               ID
<                                                                               Operator
0                                                                               Number
)                                                                               Symbol
printf                                                                          ID
(                                                                               Operator
"Error! Factorial of a negative number doesn't exist. :("                       String
)                                                                               Symbol
;                                                                               Symbol
else                                                                            KeyWord
{                                                                               Operator
for                                                                             KeyWord
(                                                                               Operator
i                                                                               ID
=                                                                               Operator
1                                                                               Number
;                                                                               Symbol
i                                                                               ID
<=                                                                              Operator
n                                                                               ID
;                                                                               Symbol
++                                                                              Operator
i                                                                               ID
)                                                                               Symbol
{                                                                               Operator
fact                                                                            ID
*=                                                                              Operator
i                                                                               ID
;                                                                               Symbol
}                                                                               Symbol
printf                                                                          ID
(                                                                               Operator
"Factorial of %d = %llu"                                                        String
,                                                                               Symbol
n                                                                               ID
,                                                                               Symbol
fact                                                                            ID
)                                                                               Symbol
;                                                                               Symbol
}                                                                               Symbol
return                                                                          KeyWord
0                                                                               Number
;                                                                               Symbol
}                                                                               Symbol
